internal/pkg/test/request.go

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+ * Copyright 2020 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package test
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+)
+
+// Marshal is a factory method that returns json-marshalled content.
+func Marshal(t *testing.T, content interface{}) []byte {
+	body, e := json.Marshal(content)
+	assert.Nil(t, e)
```

Comment:
Seems like this should be fatal, but assert isn't fatal if it fails. I think you should you check if it failed and then call t.fatal() if it did?
```
if !assert.Nil(t, e) {
  t.fatal()
}

Reply:
Point taken.  These were intended as gates to help troubleshoot an issue that should never really occur.  I'll adjust the code.